lib/Common/Memory/LargeHeapBlock.cpp

```diff
@@ -280,60 +286,93 @@ LargeHeapBlock::ReleasePagesShutdown(Recycler * recycler)
 #endif
 }
 
-template void LargeHeapBlock::ReleasePagesSweep<true>(Recycler * recycler);
-template void LargeHeapBlock::ReleasePagesSweep<false>(Recycler * recycler);
-
-template<bool pageheap>
 void
 LargeHeapBlock::ReleasePagesSweep(Recycler * recycler)
 {
     recycler->heapBlockMap.ClearHeapBlock(this->address, this->pageCount);
 
-    ReleasePages<pageheap>(recycler);
+    ReleasePages(recycler);
 }
 
+#ifdef RECYCLER_PAGE_HEAP
+__declspec(noinline)
+void LargeHeapBlock::VerifyPageHeapPattern()
+{
+    Assert(InPageHeapMode());
+    if (this->allocAddressEnd > this->address) // in case OOM while allocate heapBlockMap
+    {
+        byte* objectEndAddress = (byte*)this->allocAddressEnd;
+#if _M_X64
+        const uint64 pattern = 0xF0F0F0F0F0F0F0F0u;
+        uint64* alignedObjectEndAddress = (uint64*)Math::Align(((uint64)objectEndAddress), sizeof(void*));
```

Comment:
Can probably abstract this out of the ifdef if you use uintptr_t for both cases?